mlflow/protos/service.proto

```diff
@@ -576,6 +598,50 @@ message Experiment {
   repeated ExperimentTag tags = 7;
 }
 
+// DatasetInput. Represents a dataset and input tags.
+message DatasetInput {
+  // A list of tags for the dataset input, e.g. a “context” tag with value “training”
+  repeated InputTag tags = 1;
+
+  // The dataset being used as a Run input.
+  optional Dataset dataset = 2 [(validate_required) = true];
+}
+
+// Tag for an input.
+message InputTag {
+  // The tag key.
+  optional string key = 1 [(validate_required) = true];
+  // The tag value.
+  optional string value = 2 [(validate_required) = true];
+}
+
+// Dataset. Represents a reference to data used for training, testing, or evaluation during
+// the model development process.
+message Dataset {
```

Comment:
Can we also indicate that these objects are experimental and can be removed without warning?